dips/dip-168.md

```diff
@@ -0,0 +1,114 @@
+
+1. `CRSN(k).slots[m]` is set to `true`.
+2. `CRSN(k).min_nonce = CRSN(k).min_nonce + j`
+3. for all `j <= i < k, ``CRSN(k).slot[i - j] = CRSN(k).slot[i]` and, for all `k - j <= i < k, ``CRSN(k).slot[i] = false`.
+
+It is important to note that the sequence nonce assigned to the `force_shift(j)` transaction must be accepted by the Acceptance and Rejection Criteria in the pre-state in order for these changes to be performed.
+
+## Transaction Prioritization & Validation
+
+A transaction will only be allowed into mempool if the transaction's sequence nonce is greater than or equal to the sending account's current `min_nonce`. When validating a transaction immediately before execution, the transaction will be discarded if the sequence nonce provided with the transaction is not accepted according to the Acceptance and Rejection Criteria. With exception of the addition of the CRSN logic and checks determining if the account is using a sequence number or CRSN, transaction validation both when entering the mempool, and before execution is unchanged.
+
+After a transaction using a CRSN has entered mempool, there must be a prioritization mechanism for it when filling a block. Note that while the prioritization criteria are not part of the specification of CRSNs, the following amended rules is the most likely of these for accounts that have opted-in to using a CRSN:
+
+1. Transactions should first be ordered by gas price and sending account role as they are today
+2. Amongst equal transactions according to the ordering in (1) ties should be broken for smaller sequence nonces greater than the sending account's current `min_nonce` .
+3. Transaction's with a sequence nonce less than the sending account's current `min_nonce` should be cleared from the mempool. A transaction can only enter the state where it needs to be cleared due to force expiration.
+
+# CRSNs Satisfy Requirements
+
+1. **Non-Replayability:** If two transactions are sent from the same account `A` with the same sequence nonce `i` then the latter will be rejected by (3b), or `i` will be less than `A.CRSN(k).min_nonce` and will be rejected by (2).
+2. **Concurrency:** Transactions from the same account can be processed concurrently as long as their slot is open in the current state `A.CRSN(k)`.
+3. **Force Expiration:** A transaction with sequence nonce `i` can only be expired by processing another transaction with the same sequence nonce. A transaction can't be accidentally expired since the window may only move over filled slots (i.e., processed transactions).
```

Comment:
Does the force_shift(j) violate this property?

Reply:
Good callout. I'll clarify here that "... with the same sequence nonce, or by a `force_shift` operation is performed.", and "... the window may only move over filled slot (i.e., processed transaction) unless a specific `force_shift` operation is performed by the sender."